README.md

```diff
@@ -1 +1,33 @@
-# QuantumNLDiffEq.jl
\ No newline at end of file
+# QuantumNLDiffEq.jl
+
+### Installation
+
+```julia
+]add add YaoBlocks#master https://github.com/VarLad/QuantumNLDiffEq.jl
+```
+
+### Usage
+
+```julia
+
+julia> begin
+           using Yao, QuantumNLDiffEq
+           M = range(start=0; stop=0.9, length=20)
+           cost = sum([put(6, i=>Z) for i in 1:6])
+           var = dispatch(EasyBuild.variational_circuit(6,5), :random)
+           quantum_feature_map_circuit = chain(6, [put(i=>Ry(0)) for i in 1:6])
+           params = parameters(var)
+           u_0 = 1.0
+           N = 6
+       end
+       
+julia> train!(quantum_feature_map_circuit, cost, var, M, N, 1.0, params)
```

Comment:
This runs in rougly 7.5 minutes on my machine. I forgot that `steps=300` by default, so anyone running this example may need to be warned.

Either:
```suggestion
julia> train!(quantum_feature_map_circuit, cost, var, M, N, 1.0, params; steps=300)
```

or 

```suggestion
# WARNING: This may take several minutes
julia> train!(quantum_feature_map_circuit, cost, var, M, N, 1.0, params)
```


---

This may be out of the scope of this PR, but where could a progress bar be added?

1.  Add a `@showprogress` macro from [ProgressMeter.jl](https://github.com/timholy/ProgressMeter.jl) to the training loop in `train!`. Would this be a bloated dependency, @ChrisRackauckas?
2. Add some callback (`cb`) as a kwarg to `train!`.

Reply:
> Add a @showprogress macro from [ProgressMeter.jl](https://github.com/timholy/ProgressMeter.jl) to the training loop in train!. Would this be a bloated dependency, @ChrisRackauckas?
> Add some callback (cb) as a kwarg to train!.

Open an issue. Let's get a lean PR in first.